### ISSUE TITLE
Make `SubprocessError` as an error instance

### ISSUE THREAD

**Issue by LitoMore** [sindresorhus/nano-spawn#82]
How about make the `SubprocessError` as an error instance instead of a type?

Currently it's a bit anonying when writing the error handling code in TypeScript:

```ts
try {
	await spawn(/* ... */);
} catch (error} {
	if (error instanceof Error && (error as SubprocessError).signalName === "SIGKILL") {
		// ...
	}
}
```

The code can be cleaner if we make the `SubprocessError` as an error instnace:

```ts
try {
	await spawn(/* ... */);
} catch (error} {
	if (error instanceof SubprocessError && error.signalName === "SIGKILL") {
		// ...
	}
}
```

Please correct me if there is a better way to write the code.

**Comment by ehmicky** [comment]
Hi @LitoMore,

Thanks for the suggestion.
It makes sense. :+1:

**Comment by ehmicky** [comment]
Released with [`0.2.0`](https://github.com/sindresorhus/nano-spawn/releases/tag/v0.2.0).
